### ISSUE TITLE
Migrate to numpy>=2.0.0

### ISSUE THREAD

**Issue by WangYuNeng** [SALib/SALib#625]
Numpy 2.0.0 was released at June 16. It seems that SALib is incompatible with the new numpy now.

E.g., the quick start example, if run this after `pip install SALib --force-reinstall`
```python
from SALib.sample import saltelli
from SALib.analyze import sobol
from SALib.test_functions import Ishigami
import numpy as np

problem = {
  'num_vars': 3,
  'names': ['x1', 'x2', 'x3'],
  'bounds': [[-np.pi, np.pi]]*3
}

# Generate samples
param_values = saltelli.sample(problem, 1024)

# Run model (example)
Y = Ishigami.evaluate(param_values)

# Perform analysis
Si = sobol.analyze(problem, Y, print_to_console=True)
# Returns a dictionary with keys 'S1', 'S1_conf', 'ST', and 'ST_conf'
# (first and total-order indices with bootstrap confidence intervals)
```

will yield the following error message,
```
Traceback (most recent call last):
  File "...", line 19, in <module>
    Si = sobol.analyze(problem, Y, print_to_console=True)
  File "python3.9/site-packages/SALib/analyze/sobol.py", line 151, in analyze
    S["S1"][j] = first_order(A, AB[:, j], B)
  File ".../python3.9/site-packages/SALib/analyze/sobol.py", line 215, in first_order
    if y.ptp() == 0:
AttributeError: `ptp` was removed from the ndarray class in NumPy 2.0. Use np.ptp(arr, ...) instead.
```

Any plan on migrating to numpy>=2.0.0?

**Comment by tupui** [comment]
The is for the report. As a SciPy maintainer, this is my fault and should have known 😅 sorry about that.

I will try to get some time but if there is appetite feel free to go ahead and make a PR.

**Comment by StephanePloix** [comment]
Since I updated to numpy2.0, I get the following error:

AttributeError: module 'numpy' has no attribute 'NAN'. Did you mean: 'nan'?

when calling: results: dict = SALib.analyze.morris.analyze(problem, parameter_value_sets, numpy.array(errors, dtype=float), conf_level=0.95, print_to_console=True, num_levels=number_of_levels)

**Comment by tupui** [comment]
Just for visibility I am reopening

**Comment by tupui** [comment]
I released now 1.5.1 😃 Sorry for the delay.
